src/hooks/useImmediateEffect.ts

```diff
@@ -1,22 +1,19 @@
 /* eslint-disable react-hooks/exhaustive-deps */
 
-import { useRef, useEffect } from 'react';
-import { noop } from '../utils';
+import { useRef, useEffect, EffectCallback } from 'react';
 
 enum LifecycleState {
   WillMount = 0,
   DidMount = 1,
   Update = 2,
 }
 
-type Effect = () => () => void;
-
 /** This is a drop-in replacement for useEffect that will execute the first effect that happens during initial mount synchronously */
 export const useImmediateEffect = (
-  effect: Effect,
+  effect: EffectCallback,
   changes: ReadonlyArray<any>
 ) => {
-  const teardown = useRef(noop);
+  const teardown = useRef<ReturnType<EffectCallback>>(undefined);
```

Comment:
Are we initializing this to `undefined` with the expectation that `teardown.current` will never get called until it gets reassigned to the return type of `effect`? Seems like a safe assumption based on how this effect runs (`teardown` synchronously getting re-assigned in the first `if` below) but just wanted to get the reasoning on this change.

Reply:
Yea, should be safe since teardown functions can be undefined in the worst case 👍